File: Task_3/task_3.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program
{
    // Функция для проверки, состоит ли число из цифр, идущих по возрастанию
    static bool IsIncreasingDigits(int number)
    {
        int prevDigit = 10; // Начальное значение больше любой цифры (0-9)
        int digitCount = 0; // Счетчик цифр в числе

        while (number > 0)
        {
            int digit = number % 10;
            digitCount++;

            if (digitCount > 1 && digit >= prevDigit)
            {
                return false;
            }

            prevDigit = digit;
            number /= 10;
        }

        // Если число состоит из одной цифры, возвращаем false
        return digitCount > 1;
    }

    // Функция для переворота числа
    static int ReverseNumber(int number)
    {
        int reversed = 0;

        while (number > 0)
        {
            int digit = number % 10;
            reversed = reversed * 10 + digit;
            number /= 10;
        }

        return reversed;
    }

    static void Main()
    {
        Console.WriteLine("Введите последовательность чисел через пробел:");

        // Чтение ввода и преобразование в массив чисел
        string input = Console.ReadLine();
        var numbers = input
            .Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries)
            .Select(s =>
            {
                int result;
                return int.TryParse(s, out result) ? result : (int?)null;
            })
            .Where(n => n.HasValue)
            .Select(n => n.Value)
            .ToList();

        // Обработка чисел
        List<int> result = new List<int>();

        foreach (var number in numbers)
        {
            if (IsIncreasingDigits(number))
            {
                result.Add(ReverseNumber(number));
            }
        }

        // Вывод результата
        if (result.Count == 0)
        {
            Console.WriteLine("Искомых чисел не найдено!");
        }
        else
        {
            Console.WriteLine($"Результат: {string.Join(" ", result)}");
        }
    }
}

File: Task_3/task_3.rs
use std::io::{self, Write};

fn is_increasing_digits(mut number: u32) -> bool {
    let mut prev_digit = 10; // Начальное значение больше любой цифры (0-9)
    let mut digit_count = 0;

    while number > 0 {
        let digit = number % 10;
        digit_count += 1;

        if digit >= prev_digit {
            return false;
        }
        prev_digit = digit;
        number /= 10;
    }

    digit_count > 1
}

fn reverse_number(mut number: u32) -> u32 {
    let mut reversed = 0;

    while number > 0 {
        let digit = number % 10;
        reversed = reversed * 10 + digit;
        number /= 10;
    }

    reversed
}

fn main() {
    println!("Введите последовательность чисел через пробел:");

    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let numbers: Vec<u32> = input
        .trim()
        .split_whitespace()
        .filter_map(|s| s.parse::<u32>().ok())
        .collect();

    let mut result: Vec<u32> = Vec::new();

    for &number in &numbers {
        if is_increasing_digits(number) {
            result.push(reverse_number(number));
        }
    }

    if result.is_empty() {
        println!("Искомых чисел не найдено!");
    } else {
        println!(
            "Результат: {}",
            result.iter().map(|n| n.to_string()).collect::<Vec<_>>().join(" ")
        );
    }
}
